batch/batches/StorageUpdate/KAsyncStorageUpdate.php

```diff
@@ -63,7 +65,16 @@ public function run($jobs = null)
 				KalturaLog::debug( 'Looping '. count($partners->objects) .' partners' );
 				foreach($partners->objects as $partner)
 				{
-					$this->handlePartner($partner, KAsyncStorageUpdateUtils::PARTNER_PACKAGE_FREE);
+					if (!$maxPartner || ($partner->id <= $maxPartner))

```

Comment:
In this case we are using count for different matters than the case.
you should check if max number had reached and if so - 
1. Add printout - "Finished handling partners from XXX to YYY" 
2. Raise a flag - MaxPartnerReached and check it in the while loop.